Fix unclickable elements on iOS 18

danieldaquino opened the pull request:
This PR fixes https://github.com/damus-io/damus/issues/2370 and https://github.com/damus-io/damus/issues/2342.

The introduction of iOS 18 brought a new bug that made `KFAnimatedImage`
not recognize tap gestures and become unclickable. (https://github.com/onevcat/Kingfisher/issues/2295)

This commit addresses the issue with a workaround found here:
https://github.com/onevcat/Kingfisher/issues/2046#issuecomment-1554068070

The workaround was suggested by the author of the library to fix a
slightly different issue, but that property seems to work for our
purposes.

The issue is addressed by adding a `contentShape` property to usages
of `KFAnimatedImage`, in order to make them clickable. A custom modifier
was created to make the solution less obscure and more obvious.

Furthermore, one empty tap gesture handler was removed as it was
preventing other tap gesture handlers on the image carousel from being
triggered on iOS 18

Testing
-------

**PASS**

**Configurations:**
- iPhone 13 mini on iOS 18.0
- iPhone SE simulator on iOS 17.5

**Damus:** This commit
**Coverage:**
- Check that the following views are clickable:
    - Images in the carousel
    - Profile picture on notes
    - Profile picture on thread comments
    - Profile picture on profile page

Changelog-Fixed: Fix items that became unclickable on iOS 18
Closes: https://github.com/damus-io/damus/issues/2342
Closes: https://github.com/damus-io/damus/issues/2370
Signed-off-by: Daniel D’Aquino <daniel@daquino.me>

jb55 commented:
> The latest version of Kingfisher (8.0.2) is not compatible with our implementation, and not guaranteed to have a fix

how hard is it to update?

danieldaquino commented:
> > The latest version of Kingfisher (8.0.2) is not compatible with our implementation, and not guaranteed to have a fix
> 
> how hard is it to update?

Unfortunately after this comment I found there is no use for upgrading to 8.0.2. 

I made a minimal SwiftUI project with Kingfisher 8.0.2 and reproduced the same issue there, so it hasn't been fixed by Kingfisher.

I will probably open an issue on their repo soon, but for now I am investigating the [video control issue](https://github.com/damus-io/damus/issues/2458), since we don't even have a provisory fix there yet

danieldaquino commented:
@jb55, I updated the PR with a different approach following a recommendation from the author of Kingfisher (https://github.com/onevcat/Kingfisher/issues/2046#issuecomment-1554068070), which I have tested and feels robust. I also found a separate root cause with the carousel issue which I also fixed.

I opened an issue on the Kingfisher repo, with hopes that this can be fixed within Kingfisher itself: https://github.com/onevcat/Kingfisher/issues/2295
